mne_icalabel/gui/_label_components.py

```diff
@@ -0,0 +1,216 @@
+# -*- coding: utf-8 -*-
+"""ICA GUI for labeling components."""
+
+# Authors: Adam Li <adam2392@gmail.com>
+#
+# License: BSD (3-clause)
+
+
+import platform
+
+from matplotlib.backends.backend_qt5agg import FigureCanvas
+from matplotlib.figure import Figure
+from mne.preprocessing import ICA
+from mne.viz.backends.renderer import _get_renderer
+from qtpy import QtGui
+from qtpy.QtCore import Signal, Slot
+from qtpy.QtWidgets import (
+    QAbstractItemView,
+    QComboBox,
+    QGridLayout,
+    QHBoxLayout,
+    QLabel,
+    QListView,
+    QMainWindow,
+    QMessageBox,
+    QPlainTextEdit,
+    QPushButton,
+    QSlider,
+    QVBoxLayout,
+    QWidget,
+)
+
+# _IMG_LABELS = [['I', 'P'], ['I', 'L'], ['P', 'L']]
+# _CH_PLOT_SIZE = 1024
+# _ZOOM_STEP_SIZE = 5
+# _RADIUS_SCALAR = 0.4
+# _TUBE_SCALAR = 0.1
+# _BOLT_SCALAR = 30  # mm
+_CH_MENU_WIDTH = 30 if platform.system() == "Windows" else 10
+
+
+def _make_topo_plot(width=4, height=4, dpi=300):
+    """Make subplot for the topomap."""
+    fig = Figure(figsize=(width, height), dpi=dpi)
+    canvas = FigureCanvas(fig)
+    ax = fig.subplots()
+    fig.subplots_adjust(bottom=0, left=0, right=1, top=1, wspace=0, hspace=0)
+    ax.set_facecolor("k")
+    # clean up excess plot text, invert
+    ax.invert_yaxis()
+    ax.set_xticks([])
+    ax.set_yticks([])
+    return canvas, fig
+
+
+def _make_ts_plot(width=4, height=4, dpi=300):
+    """Make subplot for the component time-series."""
+    fig = Figure(figsize=(width, height), dpi=dpi)
+    canvas = FigureCanvas(fig)
+    ax = fig.subplots()
+    fig.subplots_adjust(bottom=0, left=0, right=1, top=1, wspace=0, hspace=0)
+    ax.set_facecolor("k")
+    # clean up excess plot text, invert
+    ax.invert_yaxis()
+    ax.set_xticks([])
+    ax.set_yticks([])
+    return canvas, fig
+
+
+def _make_spectrum_plot(width=4, height=4, dpi=300):
+    """Make subplot for the spectrum."""
+    fig = Figure(figsize=(width, height), dpi=dpi)
+    canvas = FigureCanvas(fig)
+    ax = fig.subplots()
+    fig.subplots_adjust(bottom=0, left=0, right=1, top=1, wspace=0, hspace=0)
+    ax.set_facecolor("k")
+    # clean up excess plot text, invert
+    ax.invert_yaxis()
+    ax.set_xticks([])
+    ax.set_yticks([])
+    return canvas, fig
```

Comment:
I think I would define each figure as a widget inheriting from `QtWidgets.QWidget`. Very nice find for `FigureCanvas`, I was not familiar with it and it is a class inheriting from `QWidget`. So how about:

```
class TopomapFig(FigureCanvas):
    """Topographic map widget."""
    
    def __init__(self, width=4, height=4, dpi=300):
        fig = Figure(figsize=(width, height), dpi=dpi)
        ax = fig.subplots()
        fig.subplots_adjust(bottom=0, left=0, right=1, top=1, wspace=0, hspace=0)
        ax.set_facecolor("k")
        # clean up excess plot text, invert
        ax.invert_yaxis()
        ax.set_xticks([])
        ax.set_yticks([])
        super().__init__(fig)
```

The asset is that you can now define methods for each plot, like right-click interaction, zoom, scrolling, ... in the widget class.